.circleci/config.yml

```diff
@@ -300,6 +300,7 @@ jobs:
               export CACHE_DIR=".test_cache/<< parameters.cache-dir >>"
             fi
             echo "pytest cache dir: $CACHE_DIR"
+            source pyodide_env.sh
```

Comment:
Why do we need to source this now?

Reply:
Since this PR makes os.environ no longer being updated, the PATH value will not be updated either, so we need to tell it where to find the emscripten compiler.

It is worth noting that previously pyodide-build tests relied on some implicit behavior of _get_pyodide_root()_ to update PATH:

1. _get_pyodide_root()_ calls _init_environment()_ which calls _get_make_environment_vars()_
1. _get_make_environment_vars()_ will call `make -f Makefile.envs .output_vars` which runs _pyodide_env.sh_
2. _init_environment()_ [updates os.environ](https://github.com/pyodide/pyodide/blob/4a0ca4ed48ff109679df4e246e8ed79901099f37/pyodide-build/pyodide_build/common.py#L317) (this PR removes it), so tests that require emscripten compilers can run successfully.

I think this is a very error-prone, so I changed it to explicitly source `pyodide_env.sh` before running tests that require emscripten compilers. In the future, I would like to make pyodide-build install emscripten itself (https://github.com/pyodide/pyodide/issues/3287)